File: app/src/main/java/com/lalaland/ecommerce/adapters/ProductImageAdapter.java
package com.lalaland.ecommerce.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.viewpager.widget.PagerAdapter;
import androidx.viewpager.widget.ViewPager;

import com.lalaland.ecommerce.R;
import com.lalaland.ecommerce.data.models.productDetails.ProductMultimedium;
import com.lalaland.ecommerce.databinding.ProductImageItemBinding;
import com.lalaland.ecommerce.helpers.AppConstants;
import com.squareup.picasso.Picasso;

import java.util.List;

public class ProductImageAdapter extends PagerAdapter {


    private ProductImageItemBinding productImageItemBinding;
    private List<ProductMultimedium> productMultimedia;
    private String mediaDescription;


    Context context;
    LayoutInflater layoutInflater;

    public ProductImageAdapter(Context context, List<ProductMultimedium> mProductMultimedia, String mMediaDescription) {
        this.context = context;
        productMultimedia = mProductMultimedia;
        mediaDescription = mMediaDescription;
    }

    @Override
    public int getCount() {
        return productMultimedia.size();
    }

    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object object) {

        return view == object;
    }

    @NonNull
    @Override
    public Object instantiateItem(@NonNull ViewGroup container, int position) {

        layoutInflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);

        productImageItemBinding = DataBindingUtil.inflate(layoutInflater, R.layout.product_image_item, container, false);

        String imgUrl = AppConstants.PRODUCT_STORAGE_BASE_URL.concat(productMultimedia.get(position).getSource());

        Picasso
                .get()
                .load(imgUrl)
                .placeholder(R.drawable.placeholder_products)
                .into(productImageItemBinding.ivProduct);

        productImageItemBinding.tvImageDetail.setText(mediaDescription);
        if (mediaDescription == null || mediaDescription.isEmpty())
            productImageItemBinding.tvImageDetail.setVisibility(View.GONE);
        else
            productImageItemBinding.tvImageDetail.setVisibility(View.VISIBLE);

        container.addView(productImageItemBinding.getRoot());
        return productImageItemBinding.getRoot();

    }

    @Override
    public void destroyItem(View container, int position, Object object) {
        ((ViewPager) container).removeView((View) object);
    }
}

File: lalalandAndroidApp/app/src/main/java/com/lalaland/ecommerce/adapters/ProductImageAdapter.kt
package com.lalaland.ecommerce.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.viewpager.widget.PagerAdapter
import androidx.viewpager.widget.ViewPager
import com.lalaland.ecommerce.R
import com.lalaland.ecommerce.data.models.productDetails.ProductMultimedium
import com.lalaland.ecommerce.databinding.ProductImageItemBinding
import com.lalaland.ecommerce.helpers.AppConstants
import com.squareup.picasso.Picasso

class ProductImageAdapter(var context: Context, private val productMultimedia: List<ProductMultimedium>, private val mediaDescription: String?) : PagerAdapter() {
    private lateinit var productImageItemBinding: ProductImageItemBinding
    var layoutInflater: LayoutInflater? = null
    override fun getCount(): Int {
        return productMultimedia.size
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view === `object`
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        layoutInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        productImageItemBinding = DataBindingUtil.inflate(layoutInflater!!, R.layout.product_image_item, container, false)
        val imgUrl = AppConstants.PRODUCT_STORAGE_BASE_URL + productMultimedia[position].source
        Picasso
                .get()
                .load(imgUrl)
                .placeholder(R.drawable.placeholder_products)
                .into(productImageItemBinding.ivProduct)
        productImageItemBinding.tvImageDetail.text = mediaDescription
        if (mediaDescription == null || mediaDescription.isEmpty()) productImageItemBinding.tvImageDetail.visibility = View.GONE else productImageItemBinding.tvImageDetail.visibility = View.VISIBLE
        container.addView(productImageItemBinding.root)
        return productImageItemBinding.root
    }

    override fun destroyItem(container: View, position: Int, `object`: Any) {
        (container as ViewPager).removeView(`object` as View)
    }
}